components/02-elements/cookie-message/cookie-message.js

```diff
@@ -1,46 +1,33 @@
-// TO DO - Needs a refactor
-
-'use strict'
-
-const messageClose = document.querySelector('.cookie-message__close'),
-      messageTop = document.querySelector('.cookie-message--top'),
-      cookieMessage = document.querySelector('.cookie-message--cookie');
-
-function closeBar() {
-  const cookieMessage = this.parentNode.parentNode;
-
-  if (cookieMessage.classList.contains('cookie-message--top')) {
-    localStorage.setItem('cookie-message-top', 1);
-    messageTop.style.height = 0;
-    messageTop.classList.add('closed');
+'use strict';
+class CookieMessage {
+  constructor() {
+    this.messageClose  = document.querySelector('.cookie-message__close');
+    this.cookies       = document.querySelectorAll('.cookie-message');
+
+    // todo domyslnie wylaczony, jezeli local storage nie ma wpisu to wlaczamy
+    // this.setHeight();
+    this.start();
+    this.setListeners();
   }
-  else if (cookieMessage.classList.contains('cookie-message--cookie')) {
-    localStorage.setItem('cookie-message-cookie', 1);
-    cookieMessage.style.height = 0;
-    cookieMessage.classList.add('closed');
+  closeBar(el) {
+    const cookieMessage = el.currentTarget.parentElement.parentElement,
+          dataType      = cookieMessage.dataset.type;
+
+    // if (!localStorage.getItem(dataType) !== 'closed') {
+    cookieMessage.classList.remove('cookie-message--open');
+    localStorage.setItem(dataType, 'closed')
+    // }
   }
-}
-
-function setBarHeight(el) {
-  el.style.height = 'auto';
-  el.style.height = el.clientHeight + 'px';
-}
-
-if (messageTop) {
-  setBarHeight(messageTop);
-}
-
-if (cookieMessage) {
-  setBarHeight(cookieMessage);
-}
-
-window.addEventListener('resize', () => {
-  if (messageTop) {
-    setBarHeight(messageTop);
+  setListeners() {
+    this.messageClose.addEventListener('click', this.closeBar);
   }
-  if (cookieMessage) {
-    setBarHeight(cookieMessage);
+  start() {
+    this.cookies.forEach(el => {
+      const dataType = el.dataset.type;
+      if (localStorage.getItem(dataType) !== 'closed') {
+        el.classList.add('cookie-message--open')
+      }
+    });
   }
-}, true);
-
-messageClose.addEventListener('click', closeBar);
+}
+new CookieMessage();
```

Comment:
it will don't work with multiple cookie-messages in one preview but don't know what to do with that. Adjust it and handle multiple elements in one preview (is it possible to have more than one in preview) or just left it?

Reply:
if you want we have an easy trick to do that. Pick me on PM to explain

Reply:
Using different query function was sufficient. It works with multiple messages now.